src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
 // write code here
+const mainTable = document.querySelector('table');
+
+mainTable.addEventListener('click', (e) => {
+  const header = e.target.closest('th');
+
+  if (!header) {
+    return;
+  }
+
+  const headers = mainTable.querySelector('thead').querySelectorAll('th');
+  const index = [...headers].indexOf(header);
+  const tbody = mainTable.querySelector('tbody');
+  const rows = Array.from(mainTable.rows).slice(1);
+
+  rows.sort((a, b) => sortFunction(a, b, index));
+
+  tbody.innerHTML = '';
+
+  rows.forEach((row) => tbody.appendChild(row));
+});
+
+function sortFunction(a, b, index) {
+  let valueA = a.cells[index].innerText;
+  let valueB = b.cells[index].innerText;
+
+  if (!Number.isNaN(+valueA) && !Number.isNaN(+valueB)) {
+    return valueA - valueB;
```

Comment:
The conversion from string to number using the unary plus operator (`+valueA`) is correct, but the subtraction operation (`valueA - valueB`) should ensure both values are numbers. Consider explicitly converting `valueA` and `valueB` to numbers before performing the subtraction to avoid unexpected results.